### PR TITLE
EVG-17145 go modules

### PR DESCRIPTION
**Pull request by ybrill**
[EVG-17145](https://jira.mongodb.org/browse/EVG-17145)

The changes from #62 were reverted. I think it will be hard to do development on logkeeper in its current state. This PR 
- switches to go modules again
- cherry-picks the changes to the `.evg.yml` and the like to fix tests on the waterfall.

### CODE REVIEW

**Comment by ybrill** [comment]
The changes for this PR are too large to patch (as they were in #62).
I merged and reverted the changes so I could use the initial commit as a basis for patching any additional changes in this PR. The changes from the PRs initial commit are tested in [this version](https://spruce.mongodb.com/version/logkeeper_4d9f64df51922e10b552d3ef648a871e0b071316/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC).

File: log.go [hunk 1]
```diff
@@ -99,22 +100,13 @@ func GetSender(queue amboy.Queue, fn string) (send.Sender, error) {
 	if splunk := send.GetSplunkConnectionInfo(); splunk.Populated() {
 		sender, err = send.NewSplunkLogger(name, splunk, send.LevelInfo{Default: level.Info, Threshold: level.Info})
 		if err != nil {
-			return nil, errors.Wrap(err, "problem creating the splunk logger")
+			return nil, errors.Wrap(err, "creating the splunk logger")
```
Comment (julianedwards) [inline log.go:103]: `Splunk`

File: log.go [hunk 2]
```diff
@@ -99,22 +100,13 @@ func GetSender(queue amboy.Queue, fn string) (send.Sender, error) {
...
-
-		senders = append(senders, send.NewBufferedSender(sender, interval, bufferCount))
-	}
-
-	if endpoint := os.Getenv("GRIP_SUMO_ENDPOINT"); endpoint != "" {
-		sender, err = send.NewSumo(name, endpoint)
+		bufferedSender, err := send.NewBufferedSender(ctx, sender, send.BufferedSenderOptions{FlushInterval: interval, BufferSize: bufferCount})
 		if err != nil {
-			return nil, errors.Wrap(err, "problem creating the sumo logic sender")
+			return nil, errors.Wrap(err, "creating the buffered splunk logger")
```
Comment (julianedwards) [inline log.go:107]: `Splunk`

**Review by julianedwards — APPROVED** [review]
LGTM % nits

File: log.go (hunk 1 above)
Comment (ybrill) [reply log.go]: done

File: log.go (hunk 2 above)
Comment (ybrill) [reply log.go]: done

**Comment by ybrill** [comment]
[Patch](https://spruce.mongodb.com/version/62c5e0dfa4cf4744acd75d50/tasks) with the new commits (up to and including [30bcd7c](https://github.com/evergreen-ci/logkeeper/pull/67/commits/30bcd7c592df322630c41e3e867247a9651e77c5))

**Comment by ybrill** [comment]
[Patch](https://spruce.mongodb.com/version/62c5e24ac9ec4443ed3ab432/tasks) with [ee29221](https://github.com/evergreen-ci/logkeeper/pull/67/commits/ee29221764732cfc8edbaaa14c5d9feb8949de69)

**Pull request merged**